abci/client/socket_client.go

```diff
@@ -207,12 +209,15 @@ func (cli *socketClient) didRecvResponse(res *types.Response) error {
 	reqres.Done()            // Release waiters
 	cli.reqSent.Remove(next) // Pop first item from linked list
 
-	// Notify reqRes listener if set
+	// Notify reqRes listener if set (request specific callback).
+	// NOTE: it is possible this callback isn't set on the reqres object
+	// at this point, in which case it will be called after, when it is set.
+	// TODO: should we move this after the resCb call so the order is always consistent?
```

Comment:
consistent with a localclient, you mean?

Reply:
Yes exactly. For remote client the reqResCb happens first, for local client it happens second

Reply:
follow up in https://github.com/tendermint/tendermint/issues/3513